src/test/java/com/googleinterns/zoomtube/servlets/LectureServletTest.java

```diff
@@ -0,0 +1,150 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.googleinterns.zoomtube.servlets;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.googleinterns.zoomtube.data.Lecture;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Rule;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+@RunWith(JUnit4.class)
+public final class LectureServletTest {
+  private final LocalServiceTestHelper testServices =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private DatastoreService datastoreService;
+  private LectureServlet servlet;
+  
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  private static final String LINK_INPUT = "link-input";
+  private static final String TEST_LINK = "https://www.youtube.com/watch?v=wXhTHyIgQ_U";
+
+  @Before
+  public void setUp() throws ServletException {
+    testServices.setUp();
+    datastoreService = DatastoreServiceFactory.getDatastoreService();
+    servlet = new LectureServlet();
+    servlet.init();
+  }
+
+  @After
+  public void tearDown() {
+    testServices.tearDown();
+  }
+
+  @Test
+  public void doPost_urlAlreadyInDatabase_shouldReturnLecture() throws IOException {
+    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
+    datastoreService.put(servlet.createLectureEntity(request));
+    
+    servlet.doPost(request, response);
+    
+    assertThat(datastoreService.prepare(new Query("Lecture")).countEntities()).isEqualTo(1);
```

Comment:
Given "Lecture" is public and will remain public (even after the refactoring) I would just use the const variable.